Userland/Libraries/LibGfx/ImageDecoder.cpp

```diff
@@ -19,60 +20,81 @@
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<BMPImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<BMPImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<PBMImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PBMImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<PGMImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PGMImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<PPMImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<PPMImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<ICOImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<ICOImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<JPGImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<JPGImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<DDSImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<DDSImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    plugin = make<QOIImageDecoderPlugin>(data, size);
+    if (plugin->sniff())
+        return plugin;
 
-        plugin = make<QOIImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+    return {};
+}
 
-        plugin = make<TGAImageDecoderPlugin>(data, size);
-        if (plugin->sniff())
-            return plugin;
+RefPtr<ImageDecoder> ImageDecoder::try_create_for_raw_bytes(ReadonlyBytes bytes)
+{
+    OwnPtr<ImageDecoderPlugin> plugin = probe_and_sniff_for_appropriate_plugin(bytes);
+    if (!plugin)
+        return {};
+    return adopt_ref_if_nonnull(new (nothrow) ImageDecoder(plugin.release_nonnull()));
+}
+RefPtr<ImageDecoder> ImageDecoder::try_create_for_mapped_file_path(StringView path, ReadonlyBytes bytes)
+{
+    LexicalPath lexical_mapped_file_path(path);
+    auto* data = bytes.data();
+    auto size = bytes.size();
 
+    OwnPtr<ImageDecoderPlugin> plugin;
+    plugin = [](StringView extension, auto* data, auto size) -> OwnPtr<ImageDecoderPlugin> {
+        OwnPtr<ImageDecoderPlugin> plugin;
+        if (extension == "tga"sv) {
+            plugin = make<TGAImageDecoderPlugin>(data, size);
+            if (plugin->sniff())
+                return plugin;
+        }
         return {};
-    }(data, size);
+    }(lexical_mapped_file_path.extension(), data, size);
```

Comment:
This would be more consistent if it were set up the same way as `try_create_for_raw_bytes`: a separate function instead of a lambda.